lib/Transforms/ConvertSecretForToStaticFor/ConvertSecretForToStaticFor.td

```diff
@@ -0,0 +1,63 @@
+#ifndef LIB_TRANSFORMS_CONVERTSECRETFORTOSTATICFOR_CONVERTSECRETFORTOSTATICFOR_TD_
+#define LIB_TRANSFORMS_CONVERTSECRETFORTOSTATICFOR_CONVERTSECRETFORTOSTATICFOR_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def ConvertSecretForToStaticFor : Pass<"convert-secret-for-to-static-for"> {
+  let summary = "Convert secret scf.for ops to have constant bounds.";
+  let description = [{
+  Conversion for For-operation that evaluate secret bound(s) to alternative For-operation with constant bound(s).
+
+  It replaces the data-dependent bounds with an If-operation to check the bounds, and conditionally execute and yield values from the For-operation's body. Running this pass alone does not result in a data-oblivious program; we have to run the `--convert-if-to-select` pass to the resulting program to convert the secret-dependent If-operation to a Select-operation.
+
+  Example input:
+    ```mlir
+      func.func @main(%secretTensor: !secret.secret<tensor<16xi32>>, %secretLower: !secret.secret<index> {secret.secret}, %secretUpper: !secret.secret<index> {secret.secret}) -> !secret.secret<i32> {
```

Comment:
```suggestion
      func.func @main(%secretTensor: !secret.secret<tensor<16xi32>>, %secretLower: !secret.secret<index>, %secretUpper: !secret.secret<index>) -> !secret.secret<i32> {
```

Reply:
These annotations are not necessary after secret-wrap-generic, so it's not particularly relevant to the docs for this pass.

Reply:
True, I missed that while copy-pasting.